parsel/selector.py

```diff
@@ -188,6 +188,7 @@ def __init__(self, text=None, type=None, namespaces=None, root=None,
         self._tostring_method = _ctgroup[st]['_tostring_method']
 
         if text is not None:
+            text = text.replace('\x00', '')
```

Comment:
Currently this is not in a right place, at least because the type check is after this .replace call.
I wonder if we should put it to create_root_node function.

Reply:
My bad, that's been fixed